src/test/java/org/opensearch/knn/index/query/KNNQueryBuilderTests.java

```diff
@@ -307,4 +309,13 @@ private void assertSerialization(final Version version, final Optional<QueryBuil
             }
         }
     }
+
+    public void testIgnoreUnmapped() throws IOException {
```

Comment:
do we have a test for other combinations of flag values and mappedFieldType not/not null?

Reply:
I can add another case that throws the exception if ignoreUnmapped is set to null

Reply:
please let's do that